src/codegen/function.rs

```diff
@@ -226,10 +229,40 @@ pub fn declaration_futures(env: &Env, analysis: &analysis::functions::Info) -> S
     )
 }
 
-pub fn bounds(bounds: &Bounds, skip: &[char], async: bool) -> String {
+pub fn bound_to_string(bound: &Bound, async: bool) -> String {
+    use analysis::bounds::BoundType::*;
+
+    match bound.bound_type {
+        NoWrapper => {
+            format!("{}: {}", bound.alias, bound.type_str)
+        }
+        IsA(Some(lifetime)) => {
+            format!("{}: IsA<{}> + {}", bound.alias, bound.type_str, if async { "Clone + 'static".into() } else { format!("'{}", lifetime) })
+        }
+        IsA(None) => format!("{}: IsA<{}>{}", bound.alias, bound.type_str, if async { " + Clone + 'static" } else { "" }),
+        // This case should normally never happened
```

Comment:
Add an assertion here then I guess, so we know when it actually happens and can correctly handle it then :)

Reply:
Done (the `unreachable!` call that you can see below).